src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,720 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+import io
+import json
+import sys
+
+import chardet
+import javaproperties
+import yaml
+from jsondiff import JsonDiffer
+from knack.log import get_logger
+from knack.util import CLIError
+
+from ._utils import resolve_connection_string, user_confirmation
+from ._azconfig.azconfig_client import AzconfigClient
+from ._azconfig.models import (KeyValue,
+                               ModifyKeyValueOptions,
+                               QueryKeyValueCollectionOptions)
+from .feature import list_feature
+from._featuremodels import map_keyvalue_to_featureflag
+
+logger = get_logger(__name__)
+FEATURE_FLAG_PREFIX = ".appconfig.featureflag/"
+FEATURE_FLAG_CONTENT_TYPE = "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+
+
+def __compare_kvs_for_restore(restore_kvs, current_kvs):
+    # compares two lists and find those that are new or changed in the restore_kvs
+    # optionally (delete == True) find the new ones in current_kvs for deletion
+    dict_current_kvs = {(kv.key, kv.label): (kv.value, kv.content_type, kv.locked, kv.tags) for kv in current_kvs}
+    kvs_to_restore = []
+    kvs_to_modify = []
+    kvs_to_delete = []
+    for entry in restore_kvs:
+        current_tuple = dict_current_kvs.get((entry.key, entry.label), None)
+        if current_tuple is None:
+            kvs_to_restore.append(entry)
+        elif current_tuple != (entry.value, entry.content_type, entry.locked, entry.tags):
+            kvs_to_modify.append(entry)
+
+    set_restore_kvs = {(kv.key, kv.label) for kv in restore_kvs}
+    for entry in current_kvs:
+        if (entry.key, entry.label) not in set_restore_kvs:
+            kvs_to_delete.append(entry)
+
+    return kvs_to_restore, kvs_to_modify, kvs_to_delete
+
+# File <-> List of KeyValue object
+
+
+def __read_kv_from_file(file_path, format_, separator=None, prefix_to_add="", depth=None):
+    config_data = {}
+    try:
+        with io.open(file_path, 'r', encoding=__check_file_encoding(file_path)) as config_file:
+            if format_ == 'json':
+                config_data = json.load(config_file)
+                if 'FeatureManagement' in config_data:
+                    del config_data['FeatureManagement']
+            elif format_ == 'yaml':
+                for yaml_data in list(yaml.safe_load_all(config_file)):
+                    config_data.update(yaml_data)
+                if 'feature-management' in config_data:
+                    del config_data['feature-management']
```

Comment:
I think we only support json. For other types like yaml and properties, we will treat everything as normal key values.

Reply:
Ok, I can log a warning and treat them as normal KV if someone tries to import features from yaml/properties file.

---
In reply to: [356304006](https://github.com/Azure/azure-cli/pull/11467#discussion_r356304006) [](ancestors = 356304006)